src/plugins/lanplus/lanplus.c

```diff
@@ -2366,7 +2339,6 @@ ipmi_lanplus_send_payload(
 	case IPMI_PAYLOAD_TYPE_RAKP_3:
 	case IPMI_PAYLOAD_TYPE_SOL:
 		free(msg_data);
-		msg_data = NULL;
```

Comment:
Please don't remove such assignments unless immediately before the end of their visibility scope.
According to C standards, accessing an invalid pointer (not the data it points to, but the pointer itself) yields an undefined behavior. Besides that, it is impossible to tell a valid pointer from an invalid one unless it's NULL. Hence, there is [this](https://github.com/ipmitool/ipmitool/wiki/Coding-Standards#memory-allocation) rule.
This line is not __immediately__ before the end of scope, so please leave the assignment.

FYR, https://stackoverflow.com/questions/17024866/when-is-it-valid-to-access-a-pointer-to-a-dead-object

Reply:
I'm surprised by the explicit ptr = NULL after free when you can trace the code to see it's never actually checked against NULL before it's then re-allocated.

Reply:
It's not because it's checked before reallocation. It's to make sure it is NULL because otherwise you can't tell a valid pointer from an invalid one and as explained on StackOverflow, even accessing a pointer itself may lead to undefined behavior if it's invalid.